src/lib/modernExtend.ts

```diff
@@ -21,6 +23,14 @@ function getEndpointsWithInputCluster(device: Zh.Device, cluster: string | numbe
     return endpoints;
 }
 
+function validateEntityCategory(category: 'config' | 'diagnostic' | undefined, access: 'STATE' | 'STATE_GET' | 'ALL'): boolean {
+    if ((access === 'ALL' && category === 'config') || (access !== 'ALL' && category === 'diagnostic')) {
```

Comment:
This check is already done when calling `expose.withCategory`, so no need to do it here.

Reply:
Ok, I’ll remove the function then and will set it if it’s not undefined